src/Datasource/EntityTrait.php

```diff
@@ -437,12 +437,16 @@ public function toArray()
         $result = [];
         foreach ($this->visibleProperties() as $property) {
             $value = $this->get($property);
-            if (is_array($value) && isset($value[0]) && $value[0] instanceof EntityInterface) {
+            if (is_array($value)) {
                 $result[$property] = [];
                 foreach ($value as $k => $entity) {
-                    $result[$property][$k] = $entity->toArray();
+                    if (method_exists($entity, 'toArray')) {
+                        $result[$property][$k] = $entity->toArray();
+                    } else {
+                        $result[$property][$k] = $entity;
+                    }
                 }
-            } elseif ($value instanceof EntityInterface) {
+            } elseif (method_exists($value, 'toArray')) {
```

Comment:
I was thinking that if whatever object had that method it'd be automatically serializable, making this method slightly more extensible instead of making it entities only. I can change it back if you think the memory required would be too much.